dashboard/app/models/user.rb

```diff
@@ -1744,9 +1744,12 @@ def can_change_own_user_type?
     # Don't allow editing user type unless we can also edit email, because
     # changing from a student (encrypted email) to a teacher (plaintext email)
     # requires entering an email address.
+    # Don't allow editing user type for students in sections, as teacher/school owns
+    # student's data
+    return can_edit_email? && sections_as_student.empty? if student?
     # Don't allow editing user type for teachers with sections, as our validations
     # require sections to be owned by teachers.
-    can_edit_email? && (student? || sections.empty?)
+    can_edit_email? && sections.empty?
```

Comment:
Hmm... `can_edit_email?` is not a requirement for the teacher -> student case, because we don't need the person to add a cleartext email address.  Do tests still pass if we remove that condition from this line?

In fact, would this whole block of logic read better if we used a conditional to frame the two (now very different) cases?

```rb
if student? # upgrading to teacher
  # Requires ability to edit email because upgrade requires adding a cleartext email address.
  # Students in sections cannot edit user type because teacher/school owns the student's data.
  can_edit_email? && sections_as_student.empty?
else # downgrading to student
  # Teachers with sections cannot downgrade because our validations require sections
  # to be owned by teachers.
  sections.empty?
end
```

Reply:
yeah, that would read much better. i think it's assumed that `can_edit_email?` is always true for a teacher, so the extra check doesn't really matter for them